clowder-framework/clowder2
# 447 send api key to extractor

This will send an API key to extractors instead of a token.

Requires this 3.0.1 version of pyclowder: https://github.com/clowder-framework/pyclowder/pull/69

To test:
- Clone pyclowder, switch to the pyclowder branch mentioned above.
- Create a virtual env and install the local pyclowder repo to it: `pip install -e ~/Git/pyclowder`
- Run the test wordcount extractor: 
```
cd ~/Git/pyclowder/sample-extractors/wordcount
export CLOWDER_VERSION=2   
export CLOWDER_URL=http://localhost:8000/
python wordcount.py
```

In Clowder, make sure heartbeat and messages services are running. Submit a text file to wordcount e.g. `POST http://localhost:3002/api/v2/files/639b31754241665a4fc3e513/extract?extractorName=ncsa.wordcount`

If the extraction is successful, it means the API Key was successfully passed.

Resolves #447 
Resolves #518
